examples/ecs/send_and_receive_events.rs

```diff
@@ -0,0 +1,173 @@
+//! From time to time, you may find that you want to both send and receive an event of the same type in a single system.
+//!
+//! Of course, this results in an error: the borrows of [`EventWriter`] and [`EventReader`] overlap,
+//! if and only if the [`Event`] type is the same.
+//! One system parameter borrows the [`Events`] resource mutably, and another system parameter borrows the [`Events`] resource immutably.
+//! If Bevy allowed this, this would violate Rust's rules against aliased mutability.
+//! In other words, this would be Undefined Behavior (UB)!
+//!
+//! There are two ways to solve this problem:
+//!
+//! 1. Use [`ParamSet`] to check out the [`EventWriter`] and [`EventReader`] one at a time.
+//! 2. Use a [`Local`] [`ManualEventReader`] instead of an [`EventReader`], and use [`ResMut`] to access [`Events`].
+//!
+//! In the first case, you're being careful to only check out only one of the [`EventWriter`] or [`EventReader`] at a time.
+//! By "temporally" seperating them, you avoid the overlap.
+//!
+//! In the second case, you only ever have one access to the underlying  [`Events`] resource at a time.
+//! But in exchange, you have to manually keep track of which events you've already read.
+//!
+//! Let's look at an example of each.
+
+use bevy::core::FrameCount;
+use bevy::ecs::event::ManualEventReader;
+use bevy::prelude::*;
+
+fn main() {
+    let mut app = App::new();
+    app.add_plugins(MinimalPlugins)
+        .add_event::<DebugEvent>()
+        .add_systems(Update, read_and_write_different_type)
+        .add_systems(
+            Update,
+            (
+                send_events,
+                debug_events,
+                send_and_receive_param_set,
+                debug_events,
+                send_and_receive_manual_event_reader,
+                debug_events,
+            )
+                .chain(),
+        );
+    // We're just going to run a few frames, so we can see and understand the output.
+    app.update();
+    // By running for longer than one frame, we can see that we're caching our cursor in the event queue properly.
+    app.update();
+}
+
+#[derive(Event)]
+struct A;
+
+#[derive(Event)]
+struct B;
+
+// This works fine, because the types are different,
+// so the borrows of the `EventWriter` and `EventReader` don't overlap.
+// Note that these borrowing rules are checked at system initialization time,
+// not at compile time, as Bevy uses internal unsafe code to split the `World` into disjoint pieces.
+fn read_and_write_different_type(mut a: EventWriter<A>, mut b: EventReader<B>) {
+    for _ in b.read() {}
+    a.send(A);
+}
+
+/// A dummy event type.
+#[derive(Debug, Clone, Event)]
+struct DebugEvent {
+    re_emit_from_param_set: bool,
+    re_emit_from_local_event_reader: bool,
```

Comment:
Bevy calls it `send` instead of `emit`, maybe use that? Resend is also one word, which might look nicer
```suggestion
    resend_from_param_set: bool,
    resend_from_local_event_reader: bool,
```